### PR TITLE
[KARAF-4188] Add support for systemd's watchdog

### PR DESCRIPTION
**Pull request by lburgazzoli**
Added basic and optional support for [systemd watchdog](http://www.freedesktop.org/software/systemd/man/sd_notify.html)

[KARAF-4188](https://issues.apache.org/jira/browse/KARAF-4188)

### CODE REVIEW

**Comment by cornelf** [comment]
:+1:

**Comment by lburgazzoli** [comment]
@jbonofre do you think this PR could be merged ?

**Comment by jbonofre** [comment]
Merging

File: assemblies/apache-karaf/pom.xml [hunk 1]
```diff
@@ -197,6 +197,9 @@
 
                         <library>mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.activator/${servicemix.specs.version};type:=default;export:=true</library>
                         <library>mvn:org.apache.servicemix.specs/org.apache.servicemix.specs.locator/${servicemix.specs.version};type:=default;export:=true</library>
+
+                        <library>mvn:net.java.dev.jna/jna/${jna.version};type:=boot;export:=false</library>
+                        <library>mvn:net.java.dev.jna/jna-platform/${jna.version};type:=boot;export:=false</library>
                     </libraries>
```
Comment (jbonofre) [inline assemblies/apache-karaf/pom.xml]: Do we really need those libraries by default ?

File: main/src/main/java/org/apache/karaf/main/Main.java [hunk 2]
```diff
@@ -367,7 +362,37 @@ private void doMonitor() throws Exception {
     Lock getLock() {
         return lock;
     }
-    
+
+    private void watchdog() {
+        if(Boolean.getBoolean("karaf.systemd.enabled")) {
+            new Thread("Karaf Systemd Watchdog Thread") {
```
Comment (jbonofre) [inline main/src/main/java/org/apache/karaf/main/Main.java]: I would add the `karaf.systemd.enabled` property (commented) in `etc/config.properties` to document it for users.

Comment (lburgazzoli) [inline main/src/main/java/org/apache/karaf/main/Main.java]: done

File: assemblies/apache-karaf/pom.xml (hunk 1 above)
Comment (lburgazzoli) [inline assemblies/apache-karaf/pom.xml]: Required as long as the SystemdDaemon class is referenced which happens if karaf.systemd.enabled=true and WATCHDOG_USEC is defined

**Comment by jbonofre** [comment]
LGTM. Can you rebase please (or I can do it for you) ?

**Comment by lburgazzoli** [comment]
I will tomorrow morning

**Comment by jbonofre** [comment]
Awesome, thanks !

**Comment by lburgazzoli** [comment]
Rebased and squashed in a single commit

**Comment by lburgazzoli** [comment]
@jbonofre anything more to fix ?

**Comment by jbonofre** [comment]
R: @jbonofre

**Pull request closed without merging**